apps/vs-code-designer/src/app/commands/createNewCodeProject/createCodeProjectSteps/createFunction/InvokeFunctionProjectSetup.ts

```diff
@@ -0,0 +1,183 @@
+import { FunctionConfigFile } from './FunctionConfigFile';
+import { AzureWizardPromptStep } from '@microsoft/vscode-azext-utils';
+import type { IProjectWizardContext } from '@microsoft/vscode-extension';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+
+/**
+ * This class represents a prompt step that allows the user to set up an Azure Function project.
+ */
+export class InvokeFunctionProjectSetup extends AzureWizardPromptStep<IProjectWizardContext> {
+  // Hide the step count in the wizard UI
+  public hideStepCount = true;
+
+  /**
+   * Prompts the user to set up an Azure Function project.
+   * @param context The project wizard context.
+   */
+  public async prompt(context: IProjectWizardContext): Promise<void> {
+    // Set the methodName and namespaceName properties from the context wizard
+    const methodName = context.methodName;
+    const namespace = context.namespaceName;
+
+    // Define the functions folder path using the context property of the wizard
+    const functionFolderPath = context.functionFolderPath;
+
+    // Create the .cs file inside the functions folder
+    await this.createCsFile(functionFolderPath, methodName, namespace);
+
+    // Create the .csproj file inside the functions folder
+    await this.createCsprojFile(functionFolderPath, methodName);
+
+    // Generate the VS Code configuration files in the specified folder
+    const createConfigFiles = new FunctionConfigFile();
+    await createConfigFiles.prompt(context);
+  }
+
+  /**
+   * Determines whether the user should be prompted to set up an Azure Function project.
+   * @param context The project wizard context.
+   * @returns True if the user has not yet set up an Azure Function project, false otherwise.
+   */
+  public shouldPrompt(_context: IProjectWizardContext): boolean {
+    return true;
+  }
+
+  /**
+   * Creates the .cs file inside the functions folder.
+   * @param functionFolderPath The path to the functions folder.
+   * @param methodName The name of the method.
+   * @param namespace The name of the namespace.
+   */
+  private async createCsFile(functionFolderPath: string, methodName: string, namespace: string): Promise<void> {
```

Comment:
is it a normal pattern to check a template in to code like this? In a .net project this is usually done with a T4 template that is then generated at build time, but not I'm not sure what the standard pattern is for TypeScript

Reply:
Yeah, lets check if there is a way to store the template and bring it in. Not having the whole txt over there